runge_kutta.py

```diff
@@ -0,0 +1,228 @@
+import os
+
+import numpy as np
+from scipy.optimize import fsolve
+
+EXPORT_DIR_PATH = 'data/'
+
+if not os.path.exists(EXPORT_DIR_PATH):
+    os.makedirs(EXPORT_DIR_PATH)
+
+
+def func_x(x, y, z):
+    mu = 80
+    a = 5
+    return mu * y - a * x
+
+
+def func_y(x, y, z):
+    return x * z - y
+
+
+def func_z(x, y, z):
+    return 1 - x * y - z
+
+
+def ksystem(k_coeffs: np.array, *hxyz: tuple) -> list:
+    """
+    Equations system for k1, k2 coefficients of the Implicit Runge-Kutta scheme of 4th order
+    :param k_coeffs:
+            k_coeffs[0] k1x
+            k_coeffs[1] k2x
+            k_coeffs[2] k1y
+            k_coeffs[3] k2y
+            k_coeffs[4] k1z
+            k_coeffs[5] k2z
+    :param xyz:
+    :return:
+    """
+    h, xyz = hxyz[0], hxyz[1:]
+    coeff_system = [  # equation for k_coeffs[0] k1x
+        k_coeffs[0] - h * func_x(xyz[0] + k_coeffs[0] / 4 + (1 / 4 - np.sqrt(3) / 6) * k_coeffs[1],
```

Comment:
combine the code resp for coeff calculation. Generalise to the case when number of functions can be arbitrary.